okhttp/src/test/java/okhttp3/internal/http2/HttpOverHttp2Test.java

```diff
@@ -187,6 +187,53 @@ private OkHttpClient buildHttp2Client() {
         (server.getHostName() + ":" + server.getPort()));
   }
 
+  @Test public void get204Response() throws Exception {
+    MockResponse responseWithoutBody = new MockResponse();
+    responseWithoutBody.status("HTTP/1.1 204");
+    responseWithoutBody.removeHeader("Content-Length");
+    server.enqueue(responseWithoutBody);
+
+    Call call = client.newCall(new Request.Builder()
+        .url(server.url("/foo"))
+        .build());
+    Response response = call.execute();
+
+    // Body contains nothing.
+    assertThat(response.body().bytes().length).isEqualTo(0);
+    assertThat(response.body().contentLength()).isEqualTo(0);
+
+    // Content-Length header doesn't exist in a 204 response.
+    assertThat(response.header("content-length")).isNull();
+
+    assertThat(response.code()).isEqualTo(204);
+
+    RecordedRequest request = server.takeRequest();
+    assertThat(request.getRequestLine()).isEqualTo("GET /foo HTTP/1.1");
+  }
+
+  @Test public void head() throws Exception {
+    MockResponse mockResponse = new MockResponse().setHeader("Content-Length", 5);
+    mockResponse.status("HTTP/1.1 200");
+    server.enqueue(mockResponse);
+
+    Call call = client.newCall(new Request.Builder()
+        .head()
+        .url(server.url("/foo"))
+        .build());
+
+    Response response = call.execute();
+
+    // Body contains nothing.
+    assertThat(response.body().bytes().length).isEqualTo(0);
+    assertThat(response.body().contentLength()).isEqualTo(0);
+
```

Comment:
`assertThat(response.body().string()).isEqualTo("");` triggered an exception. Is this expected?
```
java.lang.IllegalStateException: closed

	at okio.RealBufferedSource.select(RealBufferedSource.kt:83)
	at okhttp3.internal.Util.readBomAsCharset(Util.kt:256)
	at okhttp3.ResponseBody.string(ResponseBody.kt:172)
	at okhttp3.internal.http2.HttpOverHttp2Test.head(HttpOverHttp2Test.java:229)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at okhttp3.OkHttpClientTestRule$1.evaluate(OkHttpClientTestRule.java:37)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Yep, you can’t request the body multiple times.